### PR TITLE
ACTUALLY FIX RESTRICTED JOBS ROLLING ANTAGONIST

### PR DESCRIPTION
**Pull request by Gandalf2k15**
Yes. This SHOULD work.

:cl:
add: You should no longer roll antagonist as a restricted role.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic.dm [hunk 1]
```diff
@@ -751,6 +751,16 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 		ruleset.restricted_roles |= ruleset.protected_roles
 	if(CONFIG_GET(flag/protect_assistant_from_antagonist))
 		ruleset.restricted_roles |= JOB_ASSISTANT
+	// SKYRAT EDIT ADDITION
+	for(var/datum/job/iterating_job in subtypesof(/datum/job))
+		if(initial(iterating_job.antagonist_restricted))
```
Comment (Zonespace27) [inline code/game/gamemodes/dynamic/dynamic.dm]: early return

File: code/game/gamemodes/dynamic/dynamic.dm [hunk 2]
```diff
@@ -751,6 +751,16 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
...
+			if(initial(iterating_job.restricted_antagonists))
+				var/list/restricted_antagonists = initial(iterating_job.restricted_antagonists)
+				if(ruleset.antag_flag in restricted_antagonists)
```
Comment (Zonespace27) [inline code/game/gamemodes/dynamic/dynamic.dm]: early return

**Comment by Ziiro** [comment]
Back on oracle we had to refactor the entire antag rolling code because of lmao NPCs. If this doesn't fix it you may want to take a look: https://github.com/OracleStation/OracleStation/pull/684

**Comment by Gandalf2k15** [comment]
> Back on oracle we had to refactor the entire antag rolling code because of lmao NPCs. If this doesn't fix it you may want to take a look: [OracleStation/OracleStation#684](https://github.com/OracleStation/OracleStation/pull/684)

Ah awesome, thanks, but this is more for Dynamic, since it's got a habit of really just ignoring what we tell it to do

**Comment by Ziiro** [comment]
Yeah, fair enough. I figure that code is pretty out of date structurally at this point, but thought it might be worth mentioning.

**Comment by hugmeorelse** [comment]
If this works, let's re-enable families.

**Comment by Zergspower** [comment]
if it was TM'd last night it isnt working

**Comment by Gandalf2k15** [comment]
> if it was TM'd last night it isnt working

Server hasn't been updated

**Comment by SpaceVampire** [comment]
This still isn't working, just for the record.
It might be best to think about returning to the old system.

**Comment by Weblure** [comment]
Definitely not working, at least for bouncers

**Comment by Gandalf2k15** [comment]
GOOD NEWS EVERYONE, I FIXED IT

**Pull request merged**